packages/ui/src/components/DatePicker/SDatePicker.vue

```diff
@@ -47,17 +47,30 @@ const props = withDefaults(defineProps<Props>(), {
 })
 
 const innerModelValue = shallowRef<ModelValueType>(props.modelValue)
+watch(
+  () => props.modelValue,
+  (value) => {
+    innerModelValue.value = value
+  },
+  { flush: 'sync' },
+)
 
 const emit = defineEmits(['update:modelValue'])
 
-const dateFilter = (d: Date) => {
-  const minDate = props.min && startOfDay(new Date(props.min))
-  const maxDate = props.max && startOfDay(new Date(props.max))
+function isEqualToMinutes(a: Date, b: Date) {
+  return isSameDay(a, b) && isSameHour(a, b) && isSameMinute(a, b)
+}
+
+const dateFilter = (d: Date, precision: 'date' | 'datetime' = 'date') => {
+  const minDate = props.min && (precision === 'date' ? startOfDay(new Date(props.min)) : new Date(props.min))
+  const maxDate = props.max && (precision === 'date' ? startOfDay(new Date(props.max)) : new Date(props.max))
```

Comment:
Consider making `minDate` and `maxDate` `computed`s in order to prevent re-computation for each filter invocation.

Reply:
Done